core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -143,6 +143,12 @@ class HistoryServer(
     appCache.stop()
   }
 
+  // For testing - override stop timeout used by jetty
+  private[history] def setStopTimeout(timeout: Long): Unit = {
```

Comment:
I think this is a little fishy, because other tests run Jetty and don't seem to suffer from this problem (or at least no one has complained).

Maybe, if you explicitly close the selenium driver in the test, this will go away without the need for this method?